tasks/hana_deploy.yml

```diff
@@ -1,18 +1,29 @@
 ---
 
-- name: Use SAPCAR to extract the SAP HANA Bundle SAR file
-  command: >-
-    {{ sap_hana_deployment_sapcar_path }}/{{ sap_hana_deployment_sapcar_file_name }} \
-    -xvf {{ sap_hana_deployment_bundle_path }}/{{ sap_hana_deployment_bundle_sar_file_name }} \
-    -manifest SIGNATURE.SMF
-  register: extractbundle
-  args:
-    chdir: "{{ sap_hana_deployment_sapcar_path }}"
-  changed_when: "'SAPCAR: processing archive' in extractbundle.stdout"
+- name: Unpack SAPCAR archive (backwards compatibility)
+  block:
+    - name: Use SAPCAR to extract the SAP HANA Bundle SAR file
+      command: >-
+        {{ sap_hana_deployment_sapcar_path }}/{{ sap_hana_deployment_sapcar_file_name }} \
+        -xvf {{ sap_hana_deployment_bundle_path }}/{{ sap_hana_deployment_bundle_sar_file_name }} \
+        -manifest SIGNATURE.SMF
+      register: sap_hana_deployment_register_extractbundle
+      args:
+        chdir: "{{ sap_hana_deployment_sapcar_path }}"
+      changed_when: "'SAPCAR: processing archive' in sap_hana_deployment_register_extractbundle.stdout"
+
+    - name: Setting fact for HANA installer path
+      set_fact:
+        sap_hana_installdir: "{{ sap_hana_deployment_sapcar_path }}/SAP_HANA_DATABASE"
+  when:
+    - not((sap_hana_deployment_bundle_path is undefined) or ( sap_hana_deployment_bundle_path is none) or (sap_hana_deployment_bundle_path | trim == '') )
+    - not((sap_hana_deployment_bundle_sar_file_name) or (sap_hana_deployment_bundle_sar_file_name) or (sap_hana_deployment_bundle_sar_file_name | trim == ''))
 
-- name: Setting fact for HANA installer path
-  set_fact:
-    hana_source_path: "{{ sap_hana_deployment_sapcar_path }}/SAP_HANA_DATABASE"
+- name: Check availability of "{{ sap_hana_installdir + '/hdblcm' }}"
```

Comment:
You are missing a conditional for this Task or this will always fail teh user does not provide a value for `sap_hana_installdir`

Reply:
Hi Mak, this shouldn't happen. If it fails due to unset sap_hana_installdir at this point, it is intended to fail. If you define and unpack the bundle it will be set vi setfact, becasue I  have replaced hana_source_path with sap_hana_installdir in line 15.
Your check doesn't reflect that
This i

Reply:
hana_source_path was used only three times in the script for exactly the same what sap_hana_installdir is for. So I decided to replace that for simplicity reasons. 

l